### DISCUSSION TITLE
How can I convert JSON responses from custom Graph API requests into PnP models?

### DISCUSSION THREAD

**Discussion by bajce** [pnp/pnpcore#1457] [Q&A]
Is it possible to convert the json batch response from custom Microsoft Graph API requests into PnP models?
I'm using a custom Graph API request to load sharing links using `drives/{driveId}/items/{itemId}/permissions?$filter=Link ne null` because I believe I can't use `item.File.GetShareLinks()` in a batch and also, there are sometimes issues with `File.GetShareLinks()` as noted in [this GitHub issue](https://github.com/pnp/pnpcore/issues/1440). 
How can I conver the JSON custom API response into an IGraphPermissionCollection model, similar to the one returned by `File.GetShareLinks()` or `Folder.GetShareLinks()`?

**Comment by jansenbe** [comment]
@bajce : we currently don't have an easy way to populate the model via a custom request as the use case for these custom API calls was areas for which the SDK does not have support. That being said, please include the issues you're seeing in the mentioned GH issue and I'll see if can get that fixed (I'm very busy ~ vacation / conference, so it might take a while).

Long term, this is something we can consider adding.
